sawmill-core/src/main/java/io/logz/sawmill/ConditionalFactoriesLoader.java

```diff
@@ -55,21 +47,32 @@ public void loadAnnotatedProcessors(ConditionFactoryRegistry conditionFactoryReg
                 conditionsLoaded++;
             } catch (Exception e) {
                 logger.error("failed to load condition {}", condition.getName(), e);
-            }
-            finally {
+            } finally {
                 timeElapsed = stopwatch.elapsed(MILLISECONDS);
             }
         }
         logger.debug("{} conditions factories loaded, took {}ms", conditionsLoaded, stopwatch.elapsed(MILLISECONDS));
     }
 
+    // TODO add test case
     public Condition.Factory getFactory(ConditionProvider conditionProvider) throws InstantiationException, IllegalAccessException, java.lang.reflect.InvocationTargetException, NoSuchMethodException {
         Class<? extends Condition.Factory> factoryType = conditionProvider.factory();
         Optional<? extends Constructor<?>> injectConstructor = Stream.of(factoryType.getConstructors())
                 .filter(constructor -> constructor.isAnnotationPresent(Inject.class)).findFirst();
         if (injectConstructor.isPresent()) {
             Class<?>[] servicesToInject = injectConstructor.get().getParameterTypes();
-            Object[] servicesInstance = Stream.of(servicesToInject).map(services::get).toArray();
+            Object[] servicesInstance = Stream.of(servicesToInject)
+                    .peek(serviceType -> {
+                        if (!services.containsKey(serviceType)) {
+                            throw new SawmillException(String.format(
+                                    "Could not instantiate %s condition, %s dependency missing",
+                                    conditionProvider.type(),
+                                    serviceType.getSimpleName()
+                            ));
+                        }
```

Comment:
let's extract to a method `verifyDependencyIsProvided`

Reply:
agree